arch: arm: nrf: Enable SEGGER RTT on all Nordic SoCs.

pizi-nordic opened the pull request:
All chips from nRF51 and nRF52 series supports Segger RTT,
so we can enable it if given series is used.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

pizi-nordic committed:
arch: arm: nrf: Enable SEGGER RTT on all Nordic SoCs

All chips from nRF51 and nRF52 series support Segger RTT,
so we can enable it if given series is used.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

carlescufi reviewed (approved):
Yep, makes sense. Does not depend on the IC variant at all.

ioannisg commented on arch/arm/soc/nordic_nrf/nrf51/Kconfig.series:
```diff
@@ -17,5 +17,6 @@ config  SOC_SERIES_NRF51X
 	select XIP
 	select HAS_CMSIS
 	select HAS_NRFX
+	select HAS_SEGGER_RTT
```
This was missing from nRF51; just to double-check that this is the indented behavior, now.

anangl reviewed (approved):
LGTM, although you could make it (almost) perfect by removing the period from the commit title.

ioannisg commented:
commit message: supports --> support

ioannisg reviewed (changes requested):
OK. A request for clarification and a minor typo in commit message

pizi-nordic committed:
samples: sysview: Update memory requirements

When we enabled SEGGER support on the nRF51 series SoCs, this sample
started to fail due to small amount of memory avaiable on some nRF51
SoCs. This commit updates the min_ram property in order to exclude
failing boards from the build.

Signed-off-by: Piotr Zięcik <piotr.ziecik@nordicsemi.no>

pizi-nordic commented:
@ioannisg: Comments addressed. Thank you.

carlescufi commented:
@ioannisg @pizi-nordic actually let me just double-check: Segger RTT support has nothing to do with the SoC, does it? but rather with the debugger IC that comes with the board? Does it still make sense to move to the SoC's Kconfig?

galak commented:
> @ioannisg @pizi-nordic actually let me just double-check: Segger RTT support has nothing to do with the SoC, does it? but rather with the debugger IC that comes with the board? Does it still make sense to move to the SoC's Kconfig?

I wondering if we should just remove the HAS_SEGGER_RTT, and let the user enable it if they desire?

pizi-nordic commented:
@carlescufi: The chip must have debugging interface compatible with Segger (like SWD).
@galak: IMHO HAS_SEGGER_RTT should be set by SoC config (as not all SoCs supports Segger). However it's actual usage should be controlled by other switch (is part is missing AFAIK).

ioannisg commented:
Then, it looks to me that the proper way to do this is to define generic SOC capabilities (if we do not have already), i.e. CPU_HAS_SWD. And then we should have BOARD_HAS_SEGGER_RTT depending on CPU_HAS_SWD, and defined as "default-y" in nRF5x boards.

Similar to what we did in #8650, @carlescufi ?

carlescufi commented:
SWD is very generic and has really nothing to do with Segger AFAIK. I actually think that this was okayish in the first place. The actual Segger RTT code is only compiled if `CONFIG_RTT_CONSOLE` is enabled, so `CONFIG_HAS_SEGGER_RTT` is just there to make sure the board has a segger-compatible chip. That said some boards support external debuggers so I am leaning towards getting rid completely of `HAS_SEGGER_RTT` and simply continuing to use `RTT_CONSOLE` as the only condition to compile-in the RTT code.

Also RTT doesn't even seem to require SWD:
>SEGGER RTT can be used with any J-Link model and any supported target processor which allows background memory access, which are Cortex-M and RX target

As described [here](https://www.segger.com/products/debug-probes/j-link/technology/about-real-time-transfer/)

carlescufi commented:
Adding @nordic-krch for additional input here, since @sjanc sees sizes increases when enabling `HAS_SEGGER_RTT` and that should not be the case IMHO

carlescufi commented:
@ioannisg @pizi-nordic are you sure that you need SWD for RTT? I don't see that in the doc I linked above

pizi-nordic commented:
Basically, the HAS_SEGGER_RTT means "is supported by SEGGER JLink". This does not mean, that chip has to have specific hardware. As all nRF5x support JLink, all of them support RTT. Similarly, the nRF24 does not support RTT as they are not supported by JLink.
